lib/logstash/inputs/couchdb_changes.rb

```diff
@@ -5,9 +5,17 @@
 require "net/http"
 require "uri"
 
-# Stream events from the CouchDB _changes URI.
-# Use event metadata to allow for upsert and
-# document deletion.
+# This CouchDB input allows you to automatically stream events from the 
+# CouchDB http://guide.couchdb.org/draft/notifications.html[_changes] URI.
+# Moreover, any "future" changes will automatically be streamed as well making it easy to synchronize
+# your CouchDB data with any target destination
+#
+# ### Upsert and delete
+# You can use event metadata to allow for upsert and document deletion.
```

Comment:
Hm.  I didn't make it very clear, but all non-delete operations are treated as upserts.  See line 204 in your PR version.